crates/sui-core/src/authority.rs

```diff
@@ -1029,10 +1050,23 @@ impl AuthorityState {
                 epoch_store.epoch(),
             );
 
+        let effects_with_events = effects.clone();
+
+        if epoch_store.epoch() >= REMOVE_EFFECTS_EPOCH {
+            // remove all events (except some events required by rosetta) from effects -
+            // fullnode will still
+            // index them, but they will not be transmitted over state sync.
```

Comment:
> fullnode will still index them
by this I guess it is indexing to SQLite? this will work for wave 2, but SQLite has been slow on reading events so I have merged changes to read events from txn effects.

So after wave 2, indexer will not depend on SQLite for events, but will still need some event storage on FN, I heard @patrickkuo has a PR that moves events from txn effects to a separate storage on RocksDB, that should work even though the events are not part of txn effects any more.